### PR TITLE
Check extension case is not child case

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

Context: [SC-1725](https://dimagi-dev.atlassian.net/browse/SC-1725)

If an extension case is also a child case of the same parent/host, then commcare-core does not treat it as an extension case. A docstring in `livequery` confirms that this is the expected behavior in HQ as well, but currently it's not.

This change is to resolve that, and align HQ behavior with commcare-core. The first commit updates a unit test from commcare-core, and subsequent commits make that test pass.

**NOTE:** Commit [d41a9a7](https://github.com/dimagi/commcare-hq/pull/32999/commits/d41a9a7701a24928d0ba8e8a12411c694b210c51) adds a DB query for every host case. This code gets used a lot (although extension cases are not used as much). I'd like a second opinion that this additional query isn't going to introduce a noticeable performance hit.

Marking risk as high because of the volume that this code handles.

Note that the base branch is not `master`. This PR builds on refactor PR https://github.com/dimagi/commcare-hq/pull/33115

## Feature Flag
N/A

### Safety story

Not a big change, but it is a change in a high-volume area of the codebase.

### Automated test coverage

Solid test coverage.

### QA Plan

No QA planned

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[SC-1725]: https://dimagi-dev.atlassian.net/browse/SC-1725?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by snopoke — CHANGES_REQUESTED** [review]
I don't think doing an additional DB query per extension case is tolerable for sync.

Instead of doing the additional query I think you should modify the `get_related_indices` SQL function. It is only used for livequery so there's no concern about impact elsewhere.

If you remove [this line](https://github.com/dimagi/commcare-hq/blob/cdc8b6708d365839a68cc9944a28f2819a862042/corehq/sql_accessors/sql_templates/get_related_indices.sql#L33) then you will get all the indices pointing to the case, not just the extension ones. Then you can sort them out in python and you won't have to query the DB again.

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -995,11 +995,24 @@ def __init__(self, case_id, owner_ids_on_phone):
...
+        #
+        # More info: https://dimagi-dev.atlassian.net/browse/SC-1725
+        parent_ids = {
+            index.referenced_id for index in self.indices_to_add
+            if index.relationship == const.CASE_INDEX_CHILD
+        }
+        return [
+            index for index in self.indices_to_add if (
+                index.relationship == const.CASE_INDEX_EXTENSION
+                and not index.referenced_id in parent_ids
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/models.py:1013]: nit
```suggestion
                and index.referenced_id not in parent_ids
```

**Review by millerdev — COMMENTED** [review]
I'm still digesting these changes. Due to the sensitivity of this code, I might feel more comfortable if a minimal PR was made that changed behavior, and the refactoring was done in a separate PR. That way they could be easily rolled back separately if one or the other introduced a problem.

**Comment by kaapstorm** [comment]
Closing in favor of PR #33131 which includes updates based on feedback on this PR, and is simpler to review.

**Pull request closed without merging**